components/apps_form/apps_form_field/select_user_option.tsx

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {components, OptionProps} from 'react-select';
+
+import {AppSelectOption} from '@mattermost/types/apps';
+import Avatar from 'components/widgets/users/avatar/avatar';
+
+import * as Utils from 'utils/utils';
+import BotBadge from 'components/widgets/badges/bot_badge';
+import GuestBadge from 'components/widgets/badges/guest_badge';
+import {isGuest} from 'mattermost-redux/utils/user_utils';
+import {imageURLForUser} from 'utils/utils';
+
+const {Option} = components;
+export const SelectUserOption = (props: OptionProps<AppSelectOption>) => {
+    const username = props.data.username;
+    let description = '';
+
+    if ((props.data.first_name || props.data.last_name) && props.data.nickname) {
```

Comment:
I think this description set is better wrapped in a function inside the component, easier to maintain in the future, easier to use with hooks, etc.

Reply:
If we also pass the component variables to the function we might even put this out of the component and save memory. An example:

```javascript
// outside the component
const getDescription = (data: ???): string => {
 ...
}

// inside the component
const description = getDescription(props.data);
```